mdx2/dxtbx_machinery.py

```
@@ -19,6 +18,61 @@ def __init__(self, expt):
         self._scan = expt.scan
         self._beam = expt.beam
 
+        # check compatibility
+        try:
+            # Check that we have exactly one panel
+            if len(expt.detector) != 1:
+                raise RuntimeError(
+                    f"Experiment detector has {len(expt.detector)} panels; mdx2 requires single-panel detectors"
+                )
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)